### PR TITLE
reference: add TiDB cluster alert rules

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

add TiDB cluster alert rules

### What is the related PR or file link(s)? 

Aligned with: https://github.com/pingcap/docs/pull/1475

### Which version does your change affect? 

v3.0, v2.1, dev

### CODE REVIEW

**Comment by liubo0127** [comment]
LGTM

File: dev/reference/alert-rules.md [hunk 1]
```diff
@@ -0,0 +1,1274 @@
...
+
+#### `TiDB_monitor_keep_alive`
+
+* 报警规则：
+
+    `increase(tidb_monitor_keep_alive_total[10m]) < 100`
+
+* 规则描述：
+
+    表示 TiDB 的进程是否仍然存在。如果在 10 分钟之内 `tidb_monitor_keep_alive_total` 增加次数少于 100，则 TiDB 的进程可能退出，此时会报警。
```
Comment (lonng) [inline dev/reference/alert-rules.md]:
```suggestion
    表示 TiDB 的进程是否仍然存在。如果在 10 分钟之内 `tidb_monitor_keep_alive_total` 增加次数少于 100，则 TiDB 的进程可能已经退出，此时会报警。
```

File: dev/reference/alert-rules.md [hunk 2]
```diff
@@ -0,0 +1,1274 @@
...
+
+#### `TiDB_monitor_time_jump_back_error`
+
+* 报警规则：
+
+    `increase(tidb_monitor_time_jump_back_total[10m]) > 0`
+
+* 规则描述：
+
+    如果 TiDB 所在机器的时间发生了回绕，则报警。
```
Comment (lonng) [inline dev/reference/alert-rules.md]: 时间回退应该容易理解一些

**Comment by lonng** [comment]
Rest LGTM (TiDB part)

File: dev/reference/alert-rules.md [hunk 3]
```diff
@@ -0,0 +1,1274 @@
...
+category: reference
+---
+
+# TiDB 集群报警规则
+
+本文介绍了 TiDB 集群中各组件的报警规则，包括 TiDB、TiKV、PD、TiDB Binlog、Node_exporter 和 Blackbox_exporter 的各报警项的规则描述及处理方法。
+
+## TiDB 报警规则
+
+本节介绍了 TiDB 组件的报警项。每条报警项根据严重级别进行分类：紧急级别、重要级别或者警告级别。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
本节介绍了 TiDB 组件的报警项。根据严重级别，报警项可分为三类，按照严重程度由高到低依次为：紧急级别、重要级别、警告级别。
```

File: dev/reference/alert-rules.md [hunk 4]
```diff
@@ -0,0 +1,1274 @@
...
+
+    如果 TiDB 中等待执行的 DDL 任务的数量大于 5，则报警。
+
+* 处理方法：
+
+    通过 `admin show ddl` 语句检查是否有耗时的 add index 操作正在执行。
+
+## PD 报警规则
+
+本节介绍了 PD 组件的报警项。每条报警项根据严重级别进行分类：紧急级别、重要级别或者警告级别。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
本节介绍了 PD 组件的报警项。根据严重级别，报警项可分为三类，按照严重程度由高到低依次为：紧急级别、重要级别、警告级别。
```

File: dev/reference/alert-rules.md [hunk 5]
```diff
@@ -0,0 +1,1274 @@
...
+    没有合适的 store 用来补副本。
+
+* 处理方法：
+
+    * 检查 store 是否空间不足。
+    * 根据 label 配置（如果有这个配置的话）来检查是否有可以补副本的 store。
+
+## TiKV 报警规则
+
+本节介绍了 TiKV 组件的报警项。每条报警项根据严重级别进行分类：紧急级别、重要级别或者警告级别。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
本节介绍了 TiKV 组件的报警项。根据严重级别，报警项可分为三类，按照严重程度由高到低依次为：紧急级别、重要级别、警告级别。
```

**Comment by yikeke** [comment]
Comments addressed. PTAL again. @lilin90

File: dev/reference/alert-rules.md [hunk 6]
```diff
@@ -0,0 +1,1274 @@
...
+    * 判断从 Pump 获取数据是否太慢：
+
+        监控 Pump handle tso 可以看每个 Pump 最近一条消息的时间，是不是有延迟特别大的 Pump，确认对应 Pump 正常运行。
+
+    * 根据 Drainer event 和 Drainer execute latency 来判断是否下游同步太慢：
+
+        * 如果 Drainer execute time 过大，则检查到目标库网络带宽和延迟，以及目标库状态。
+        * 如果 Drainer execute time 不大，Drainer event 过小，则增加 work count 和 batch 进行重试。
+
+    * 上面都不满足或者操作后没有改观，则报备开发 `support@pingcap.com` 进行处理。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    *  如果上面都不满足或者操作后没有改观，则报备开发人员 [support@pingcap.com](mailto:support@pingcap.com) 进行处理。
```

File: dev/reference/alert-rules.md [hunk 7]
```diff
@@ -0,0 +1,1274 @@
...
+
+    `increase(tidb_domain_load_schema_total{type="failed"}[10m]) > 10`
+
+* 规则描述：
+
+    TiDB 重载最新的 Schema 信息失败的总次数。如果在 10 分钟之内重载失败次数超过 10 次，则报警。
+
+* 处理方法：
+
+    参考 [`TiDB_schema_error`](#-tidb-schema-error) 的处理方法。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    参考 [`TiDB_schema_error`](#tidb-schema-error) 的处理方法。
```

File: dev/reference/alert-rules.md [hunk 8]
```diff
@@ -0,0 +1,1274 @@
...
+
+* 规则描述：
+
+    在 6 小时内 Region 上没有成功执行 GC，说明 GC 不能正常工作了。短期内 GC 不运行不会造成太大的影响，但如果 GC 一直不运行，版本会越来越多，从而导致查询变慢。
+
+* 处理方法：
+
+    1. 执行 `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME="tikv_gc_leader_desc"` 来找到 gc leader 对应的 `tidb-server`；
+    2. 查看该 `tidb-server` 的日志，grep gc_worker tidb.log；
+    3. 如果发现这段时间一直在 resolve locks（最后一条日志是“start resolve locks”）或者 delete ranges（最后一条日志是“start delete {number} ranges”），说明 GC 进程是正常的。否则需要联系 `support@pingcap.com` 帮忙处理。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    3. 如果发现这段时间一直在 resolve locks（最后一条日志是 `start resolve locks`）或者 delete ranges（最后一条日志是 `start delete {number} ranges`），说明 GC 进程是正常的。否则需要联系 [support@pingcap.com](mailto:support@pingcap.com) 帮忙处理。
```

File: dev/reference/alert-rules.md [hunk 9]
```diff
@@ -0,0 +1,1274 @@
...
+
+* 规则描述：
+
+    表明无法连接远端的 TiKV。
+
+* 处理方法：
+
+    1. 检查网络是否通畅。
+    2. 检查远端 TiKV 是否挂掉。
+    3. 如果远端 TiKV 没有挂掉，检查压力是否太大，参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 处理方法。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    3. 如果远端 TiKV 没有挂掉，检查压力是否太大，参考 [`TiKV_channel_full_total`](#tikv-channel-full-total) 处理方法。
```

File: dev/reference/alert-rules.md [hunk 10]
```diff
@@ -0,0 +1,1274 @@
...
+
+    `histogram_quantile(0.99, sum(rate(tikv_storage_engine_async_request_duration_seconds_bucket{type="snapshot"}[1m])) by (le, instance, type)) > 1`
+
+* 规则描述：
+
+    这个值偏大，表明 Raftstore 负载压力很大，可能已经卡住。
+
+* 处理方法：
+
+    参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 的处理方法。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    参考 [`TiKV_channel_full_total`](#tikv-channel-full-total) 的处理方法。
```

Comment (lilin90) [reply dev/reference/alert-rules.md]: Ref: you can test the link rule [here](https://pingcap.com/docs/v3.0/reference/configuration/tidb-server/configuration/#config) (an example).

File: dev/reference/alert-rules.md [hunk 11]
```diff
@@ -0,0 +1,1274 @@
...
+
+    `histogram_quantile(0.99, sum(rate(tikv_storage_engine_async_request_duration_seconds_bucket{type="write"}[1m])) by (le, instance, type)) > 1`
+
+* 规则描述：
+
+    这个值偏大，表明 Raft write 耗时很长。
+
+* 处理方法：
+
+    1. 检查 Raftstore 上的压力，参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 的处理方法。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    1. 检查 Raftstore 上的压力，参考 [`TiKV_channel_full_total`](#tikv-channel-full-total) 的处理方法。
```

File: dev/reference/alert-rules.md [hunk 12]
```diff
@@ -0,0 +1,1274 @@
...
+
+    `sum(rate(tikv_thread_cpu_seconds_total{name=~"raftstore_.*"}[1m])) by (instance, name) > 1.6`
+
+* 规则描述：
+
+    Raftstore 线程压力太大。
+
+* 处理方法：
+
+    参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 的处理方法。
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
    参考 [`TiKV_channel_full_total`](#tikv-channel-full-total) 的处理方法。
```

Comment (lilin90) [reply dev/reference/alert-rules.md]: Please check and update all similar links.

File: dev/reference/alert-rules.md [hunk 13]
```diff
@@ -0,0 +1,1274 @@
...
+* 规则描述：
+
+    Drainer 同步到 TiDB 的事务耗时。如果这个值过大，会影响 Drainer 同步。
+
+* 处理方法：
+
+    * 查看 TiDB 集群的状态。
+    * 查看 Drainer 日志或监控，如果是 DDL 操作导致了该问题，则忽略。
+
+## Node_Exporter 主机报警规则
```
Comment (lilin90) [inline dev/reference/alert-rules.md]:
```suggestion
## Node_exporter 主机报警规则
```

File: dev/reference/alert-rules.md [hunk 14]
```diff
@@ -0,0 +1,1274 @@
...
+
+#### `TiKV_memory_used_too_fast`
+
+* 报警规则：
+
+    `process_resident_memory_bytes{job=~"tikv",instance=~".*"} - (process_resident_memory_bytes{job=~"tikv",instance=~".*"} offset 5m) > 5*1024*1024*1024`
+
+* 规则描述：
+
+    目前没有和内存相关的 TiKV 的监控，你可以通过 node_exporter 监控集群内机器的内存使用情况。如上规则表示，如果在 5 分钟之内内存使用超过 5GB（TiKV 内存占用的太快），则报警。
```
[the remark this answers is not in the document]
Comment (lilin90) [reply dev/reference/alert-rules.md]: Please check it and update all occurrences if necessary. Thanks!

**Pull request merged**